sklearn/feature_selection/base.py

```diff
@@ -120,3 +120,77 @@ def inverse_transform(self, X):
         Xt = np.zeros((X.shape[0], support.size), dtype=X.dtype)
         Xt[:, support] = X
         return Xt
+
+
+def featurewise_scorer(score_func, **kwargs):
+    """ A wrapper function around score functions.
+
+    Parameters
+    ----------
+    score_func : callable
+        Function taking array(s) X and/or y, and returning a pair of arrays
+        (scores, pvalues) or a single array with scores. This function is also
+        allowed to take other parameters as input.
+
+    Returns
+    -------
+    scores : array-like, shape (n_features,)
+        Score values returned by the scoring function.
+    p_vals : array-like, shape (n_features,)
+        The set of p-values returned by the scoring function.
+
+    Notes
+    -----
+    This wrapper function wraps around scoring functions like `spearmanr`,
+    `pearsonr` etc. from the scipy.stats module and makes it usable for
+    feature selection algorithms like `SelectKBest`. Also, this wrapper
+    function takes the absolute value of the scores returned i.e. a score of
+    +1 is same as -1.
+
+    Example
+    -------
+    >>> from sklearn.feature_selection import featurewise_scorer, SelectKBest
+    >>> from scipy.stats import spearmanr
+    >>> from sklearn.datasets import make_classification
+    >>> X, y = make_classification(random_state=0)
+    >>> skb = SelectKBest(featurewise_scorer(spearmanr), k=10)
+    >>> skb.fit(X, y) #doctest: +ELLIPSIS, +NORMALIZE_WHITESPACE
+    SelectKBest(k=10, score_func=...)
+    >>> new_X1 = skb.transform(X)
+
+    """
+    def call_scorer(*args):
+        if len(args) == 2:
+            X = args[0]
+            y = args[1]
+            X, y = check_X_y(X, y, ('csr', 'csc'), multi_output=True)
+        else:
+            X = args[0]
+            y = None
+            X = check_array(X, ('csr', 'csc'))
+
+        scores = []
+        p_vals = []
+
+        for i in six.moves.range(X.shape[1]):
+            if y is None:
+                score_func_ret = score_func(X[:, i], **kwargs)
+            else:
+                score_func_ret = score_func(X[:, i], y, **kwargs)
+
+            if isinstance(score_func_ret, (list, tuple)):
+                score, p_val = score_func_ret
+                p_vals.append(p_val)
+            else:
+                score = score_func_ret
+                p_vals = None
```

Comment:
Do you need to set `p_vals = None`? You check whether `p_vals is not None` below, but looks like you can just check `len(p_vals) > 0`.